### PR TITLE
Fix route auto-reorder when only one route table is matched

### PR DESCRIPTION
**Pull request by marcogschmidt**
## Problem
Gloo currently sorts routes that originate from a single route table. This is not intended behavior. Gloo should sort routes only when a route delegates to multiple route tables, because in these cases we want to avoid short-circuiting (e.g. having `/foo` come before `/foo/bar`). 

## Solution
This change fixes the behavior and adds a new `alwaysSortRouteTableRoutes` setting for users who might be depending on the current behavior. If this flag is set to `true` (default is `false`), it will force Gloo to sort the routes even when they come from a single route table.

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2529

### CODE REVIEW

File: changelog/v1.2.22/fix-single-route-table-ordering.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+- type: FIX
+  description: Fix noisy logging on duplicate upgrade websocket config when using Gloo with SSL & SNI.
+  issueLink: https://github.com/solo-io/gloo/issues/2529
```
Comment (kdorosh) [inline changelog/v1.2.22/fix-single-route-table-ordering.yaml]: why is this in here?

File: install/helm/gloo/generate/values.go [hunk 2]
```diff
@@ -150,6 +150,7 @@ type Gateway struct {
 	UpdateValues                  bool               `json:"updateValues" desc:"if true, will use a provided helm helper 'gloo.updatevalues' to update values during template render - useful for plugins/extensions"`
 	ProxyServiceAccount           ServiceAccount     `json:"proxyServiceAccount" `
 	ReadGatewaysFromAllNamespaces bool               `json:"readGatewaysFromAllNamespaces" desc:"if true, read Gateway custom resources from all watched namespaces rather than just the namespace of the Gateway controller"`
+	AlwaysSortRouteTableRoutes    bool               `json:"alwaysSortRouteTableRoutes" desc:"if true, the Gateway will always sort routes originating from delegated route tables, rather than just when a route delegates to multiple route tables"`
```
Comment (kdorosh) [inline install/helm/gloo/generate/values.go]: let's include the default in the description

File: install/test/helm_test.go [hunk 3]
```diff
@@ -777,6 +777,7 @@ spec:
    fdsMode: WHITELIST
  gateway:
    readGatewaysFromAllNamespaces: false
+   alwaysSortRouteTableRoutes: false
```
Comment (kdorosh) [inline install/test/helm_test.go]: we shouldn't need this

File: changelog/v1.2.22/fix-single-route-table-ordering.yaml (hunk 1 above)
Comment (marcogschmidt) [reply changelog/v1.2.22/fix-single-route-table-ordering.yaml]: Copy paste error, fixing

**Pull request merged**